src/D2L.CodeStyle.Analyzers/Diagnostics.cs

```diff
@@ -571,5 +571,14 @@ public static class Diagnostics {
 			defaultSeverity: DiagnosticSeverity.Error,
 			isEnabledByDefault: true
 		);
+
+		public static readonly DiagnosticDescriptor AnonymousFunctionsMayCaptureMutability = new DiagnosticDescriptor(
+			id: "D2L0077",
+			title: "Anonymous functions may capture mutable state and thus can't be assigned to members of an immutable type.",
+			messageFormat: "Anonymous functions may capture mutable state and thus can't be assigned to members of an immutable type. Try making the function static, or moving the assignment to an initializer.",

```

Comment:
I don't think it's required for safety, because the set of stuff they can capture isn't mutable owing to the rest of our analysis (or if it is, there is another diagnostic... this matches the general philosophy of the analyzer)

Reply:
Yeah, I was just thinking towards not having to distinguish the two cases (yet) a la https://github.com/Brightspace/D2L.CodeStyle/pull/729#discussion_r591680018